src/main/java/net/jitl/common/entity/overworld/HonglowEntity.java

```diff
@@ -1,110 +1,156 @@
+import net.minecraft.entity.ILivingEntityData;
+import net.minecraft.entity.LivingEntity;
 import net.minecraft.entity.MobEntity;
 import net.minecraft.entity.SpawnReason;
 import net.minecraft.entity.ai.attributes.AttributeModifierMap;
 import net.minecraft.entity.ai.attributes.Attributes;
 import net.minecraft.entity.ai.goal.*;
-import net.minecraft.entity.monster.MonsterEntity;
 import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.nbt.CompoundNBT;
+import net.minecraft.network.datasync.DataParameter;
+import net.minecraft.network.datasync.EntityDataManager;
 import net.minecraft.potion.EffectInstance;
 import net.minecraft.potion.Effects;
-import net.minecraft.potion.Potion;
 import net.minecraft.util.*;
 import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.DifficultyInstance;
+import net.minecraft.world.IServerWorld;
 import net.minecraft.world.IWorld;
 import net.minecraft.world.World;
 import net.minecraft.world.biome.Biome;
-import net.minecraftforge.api.distmarker.Dist;
-import net.minecraftforge.api.distmarker.OnlyIn;
+import ru.timeconqueror.timecore.api.util.Requirements;
+
 import org.jetbrains.annotations.NotNull;
 
 import java.util.Random;
 
-public class HonglowEntity extends MonsterEntity {
+import javax.annotation.Nullable;
 
-	//=============================CLIENT STUFF=============================\\
-	Random rand = new Random();
-	private final String[] textures = {
-			"honglow_blue.png",
-			"honglow_red.png",
-			"honglow_green.png"
-	};
-	private final String finalTexture = textures[rand.nextInt(textures.length)];
-	//======================================================================\\
+public class HonglowEntity extends CreatureEntity {
+	public static final DataParameter<Type> VARIANT = EntityDataManager.defineId(HonglowEntity.class, JDataSerializers.HONGLOW_VARIANT);
 
 	public HonglowEntity(EntityType<? extends HonglowEntity> entityType, World world) {
 		super(entityType, world);
 	}
 
 	protected void registerGoals() {
 		this.goalSelector.addGoal(0, new SwimGoal(this));
-		this.goalSelector.addGoal(1, new MeleeAttackGoal(this, 0.28D, true));
 		this.goalSelector.addGoal(6, new WaterAvoidingRandomWalkingGoal(this, 1.0D));
 		this.goalSelector.addGoal(7, new LookAtGoal(this, PlayerEntity.class, 6.0F));
 		this.goalSelector.addGoal(8, new LookRandomlyGoal(this));
 	}
+	
+	public Type getVariant() {
+		return this.getEntityData().get(VARIANT);
+	}
+	
+	public Type getVariantFromInt(int number) {
+		Requirements.inRangeInclusive(number, 0, 2);
+		switch (number) {
+			case 0:
+				return Type.RED;	
+			case 1: 
+				return Type.GREEN;	
+			case 2:
+				return Type.BLUE;
+		}
+		return null;
+	}
+	
+	@Override
+	public ILivingEntityData finalizeSpawn(IServerWorld worldIn, DifficultyInstance difficultyIn, SpawnReason reason, @Nullable ILivingEntityData spawnDataIn, @Nullable CompoundNBT dataTag) {
+		
+		this.getEntityData().set(VARIANT, getVariantFromInt(this.random.nextInt(3)));
```

Comment:
if we change Variant count, this may crash or not work properly. To fix it use RandHelper.chooseEqually(Type.values())